components/director/internal/systemfetcher/director.go

```diff
@@ -0,0 +1,60 @@
+package systemfetcher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/form3tech-oss/jwt-go"
+	gcli "github.com/machinebox/graphql"
+	"github.com/pkg/errors"
+)
+
+type Authenticator interface {
+	GetAuthorization(ctx context.Context, tenant string) (string, error)
+}
+
+type DirectorGraphClient struct {
+	*gcli.Client
+	authenticator Authenticator
+}
+
+func (d *DirectorGraphClient) DeleteSystemAsync(ctx context.Context, id, tenant string) error {
+	gqlRequest := gcli.NewRequest(fmt.Sprintf(`mutation {
+			  unregisterApplication(
+				id: %q,
+				mode: ASYNC
+			  ) {
+				  id
+			    }
+			}`, id))
+	token, err := d.authenticator.GetAuthorization(ctx, tenant)
+	if err != nil {
+		return err
+	}
+
+	gqlRequest.Header.Set("Authorization", "Bearer "+token)
+
+	if err := d.Run(ctx, gqlRequest, nil); err != nil {
```

Comment:
We ought to handle gracefully cases where a delete async request is already in progress (i.e. concurrency check code in director middleware will return an error to this client here)

Reply:
At the moment it does not fail. Just prints the error returned and continues with the rest of the systems. Even if a check is added whether the operation is in progress, not sure what else we can do aside from log it.